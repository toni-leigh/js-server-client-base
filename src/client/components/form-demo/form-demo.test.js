import React from 'react';
import { createStore } from 'redux';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import {
  FormDemo,
  FormDemoConnected,
  mapDispatchToProps,
  mapStateToProps
} from './form-demo';



describe('<FormDemo />', () => {
  let formDemo;

  beforeAll(() => {
    formDemo = shallow(
      <FormDemo />
    );
  });

  test('basic dumb component render', () => {
    expect(formDemo).toMatchSnapshot();
  });
});



describe('<FormDemoConnected />', () => {
  let formDemoConnected;

  beforeAll(() => {
    const store = createStore(() => {});
    formDemoConnected = shallow(
      <FormDemoConnected store={ store } />
    );
  });

  test('render occurs through connection', () => {
    expect(formDemoConnected).toMatchSnapshot();
  });
});



describe('mapDispatchToProps', () => {
  test('processes the actions correctly', () => {
    expect(mapDispatchToProps).toMatchSnapshot();
  });
});



describe('mapStateToProps', () => {
  test('processes the state correctly', () => {
  const state = fromJS({});
    expect(mapStateToProps(state)).toMatchSnapshot();
  });
});
