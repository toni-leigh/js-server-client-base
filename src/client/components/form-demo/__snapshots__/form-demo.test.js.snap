// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormDemo /> basic dumb component render 1`] = `
<form
  className="server-client-form-demo"
>
  FormDemo
</form>
`;

exports[`<FormDemoConnected /> render occurs through connection 1`] = `
<withImmutablePropsToJS(FormDemo)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`mapDispatchToProps processes the actions correctly 1`] = `Object {}`;

exports[`mapStateToProps processes the state correctly 1`] = `Immutable.Map {}`;
